test/extensions/INTEL/SPV_INTEL_fpga_memory_attributes/IntelFPGAMemoryAttributesForStruct.ll

```diff
@@ -116,21 +87,22 @@ target triple = "spir64-unknown-unknown"
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[NUMBANKS]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[BANKWIDTH]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[MAX_PRIVATE_COPIES]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[SINGLEPUMP]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[DOUBLEPUMP]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[MAX_REPLICATES]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[SIMPLE_DUAL_PORT]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[MERGE]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[BANK_BITS]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[FORCE_POW_2_DEPTH]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[STRIDESIZE]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[WORDSIZE]], ptr undef, i32 undef, ptr undef)
-; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0.p0(ptr %{{[a-zA-Z0-9_]+}}, ptr [[TRUE_DUAL_PORT]], ptr undef, i32 undef, ptr undef)
+
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[REGISTER]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[MEMORY]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[NUMBANKS]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[BANKWIDTH]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[MAX_PRIVATE_COPIES]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[SINGLEPUMP]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[DOUBLEPUMP]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[MAX_REPLICATES]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[SIMPLE_DUAL_PORT]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[MERGE]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[BANK_BITS]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[FORCE_POW_2_DEPTH]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[STRIDESIZE]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[WORDSIZE]]
+; CHECK-LLVM: call void @llvm.var.annotation{{.*}}(ptr {{.*}}, ptr [[TRUE_DUAL_PORT]]
```

Comment:
This is because the SPIRV reverse translator relies on a heuristic to decide how to translate SPIRV `OpDecoration`. Ignoring casts, if the decoration is on an `alloca` translate to an `llvm.var.annotation` else translate to `llvm.ptr.annotation`. 
Prior to this change `OpMemberDecoration` will always map to `llvm.ptr.annotation`.